examples/guides/file-integration/ftp-listener/README.md

```diff
@@ -0,0 +1,236 @@
+# Implementing ftp listener with Ballerina
+
+This example demonstrates how implement ftp listener in Ballerina.
+
+#### What you'll build
+
+In this example, we'll implement ftp listener which will listen to a remote directory and periodically notify the file addition of the specified file pattern.
+
+Then we can define the processing we need to upon the file content.
+
+Finally we can move the processed file to the specified location or delete it.
+
+#### Prerequisites
+
+* Ballerina Distribution
+* A Text Editor or an IDE <br/>
+    Tip: For a better development experience, install one of the following Ballerina IDE plugins: VSCode, IntelliJ IDEA
+* Install ftp connectors by following the below steps.<br/>
+1. Download correct distribution.zip from releases (we can get it from the link given below) that match with ballerina version. <br/>
+https://github.com/wso2-ballerina/module-ftp<br/>
+2. Unzip package distribution.<br/>
+3. Run the install.<sh|bat> script to install the package. You can uninstall the package by running uninstall.<sh|bat>.  <br/>
+* FTP server and client  
+
+### Let's Get Started!
+
+This tutorial includes the following sections.
+
+- [Implementation](#implementation)
+  - [Creating the Project Structure](#creating-the-project-structure)
+  - [Implementing the FTP Listener](#Implementing-the-FTP-Listener)
+  - [Implementing the FTP Client](#Implementing-the-FTP-Client) 
+- [Deployment](#deployment)
+  - [Deploying Locally](#deploying-locally)
+  - [Deploying on Docker](#deploying-on-docker)
+- [Testing](#testing)
+  - [Invoking the Database Service](#invoking-the-database-service)
+
+
+### Implementation
+
+#### Creating the project structure
+
+Ballerina is a complete programming language that supports custom project structures. Use the following package structure for this guide.
+
+```
+  └──file-integration
+    └── ftp-listener
+        └── ftp_listener.bal            
+```
+        
+Create the above directories in your local machine and also create empty .bal files.
+
+Then open the terminal and navigate to ftp-template and run Ballerina project initializing toolkit.
+
+```ballerina
+   $ ballerina init
+```
+
+#### Implementing the FTP Listener
```

Comment:
Shall we change the flow of this example and highlight the things that the user needs to change after auto-generating the code using templates or snippets. Maybe give guides on how to get this boilerplate code auto-generated.